### PR TITLE
Adding threshold to Add And Enable button

### PR DESCRIPTION
**Pull request by corrigancd**
So the changes are:
- The add layers button is now filled while the add and enable button is empty (it used to be the other way around)
- If there are more than 35 layer checkboxes ticked (i.e. the performance threshold and to avoid browser crash), the Add and Enable button will be greyed out. I have tested this with a threshold of 4 as per GIF

GreyedOutAddLayersButton

### CODE REVIEW

File: public/lib/dragAndDroplayercontrol/layerContolTree.js [hunk 1]
```diff
@@ -339,18 +342,30 @@ export class AddMapLayersModal extends React.Component {
         </EuiFlexItem>
 
         <EuiFlexItem grow={false}>
-          <EuiButton
+          {this.state.selectedLayerCount > 35 && <EuiButton
```
Comment (corrigancd) [inline public/lib/dragAndDroplayercontrol/layerContolTree.js:345]: @niknbr , just wondering there is a cleaner way to adding the isDisabled flag in this type of case?

Comment (szydan) [reply public/lib/dragAndDroplayercontrol/layerContolTree.js:345]: There is "isDisabled" property on EUI button
Can we do 
```javascript
isDisabled={this.state.selectedLayerCount <= 35}
```

**Comment by johnnylopes** [comment]
Please, we also need a warning message on the modal saying why the button is disabled.

**Comment by corrigancd** [comment]
@johnnylopes , how about this. I've set the limit to 35 in the actual PR (as this was the threshold that performance was identified to be poor beyond):

image

**Comment by corrigancd** [comment]
> not sure how this would look
> but maybe we could swap the buttons as well

@szydan , this one better? 
image

**Comment by johnnylopes** [comment]
thanks, @corrigancd.

I would like someone else to review it (@jccq @szydan  )..it seems a big text for that small modal. Is this going to be on top of the layers like that?

If this information is going to be present on the documentation, maybe we could put a small warning and a link to the documentation?

**Comment by louise-glavey** [comment]
@corrigancd I suggest that you modify the message as follows:
**Adding multiple layers?**
When more than four layers are selected, only the **Add** option is available. 
To make the layers visible after they are added, select the individual checkbox for each layer in Layer Control.

**Comment by corrigancd** [comment]
Here is how things look, with @louise-glavey feedback:

image

**Comment by jccq** [comment]
Too verbose can we pls have

Title "Too many layers to activate right away"
text: "You can add them now and activate them later in the Layer Control"

**Pull request merged**